docs/additional-functionality/qualification-profiling-tools.md

```diff
@@ -0,0 +1,708 @@
+GPU generated event logs.
+
+(The code is based on Apache Spark 3.1.1 source code, and tested using Spark 3.0.x and 3.1.1 event logs)
+
+This document covers below topics:
+
+* [Prerequisites](#Prerequisites)
+* [Download the tools jar or compile it](#Download-the-tools-jar-or-compile-it)
+* [Qualification Tool](#Qualification-Tool)
+    * [Qualification tool functions](#Qualification-tool-functions)
+    * [Download the Spark 3 distribution](#Download-the-Spark-3-distribution)
+    * [How to use qualification tool](#How-to-use-qualification-tool)
+    * [Qualification tool options](#Qualification-tool-options)
+    * [Qualification tool output](#Qualification-tool-output)
+* [Profiling Tool](#Profiling-Tool)
+    * [Profiling tool functions](#Profiling-tool-functions)
+    * [Profiling tool metrics definitions](#Profiling-tool-metrics-definitions)
+    * [How to use profiling tool](#How-to-use-profiling-tool)
+    * [Profiling tool options](#Profiling-tool-options)
+    * [Profiling tool output](#Profiling-tool-output)
+      
+## Prerequisites
+- Spark 3.0.1 or newer, the Qualification tool just needs the Spark jars and the Profiling tool
+  runs a Spark application so needs the Spark runtime.
+- Java 8 or above
+- Spark event log(s) from Spark 2.0 or above version.
+  Support both rolled and compressed event logs with `.lz4`, `.lzf`, `.snappy` and `.zstd` suffixes.
+  Also support Databricks specific rolled and compressed(.gz) eventlogs.
+  The tool does not support nested directories, event log files or event log directories should be
+  at the top level when specifying a directory.
+
+Note: Spark event logs can be downloaded from Spark UI using a "Download" button on the right side,
+or can be found in the location specified by `spark.eventLog.dir`. See the
+[Apache Spark Monitoring](http://spark.apache.org/docs/latest/monitoring.html) documentation for
+more information.
+
+Optional:
+- maven installed 
+  (only if you want to compile the jar yourself)
+- hadoop-aws-<version>.jar and aws-java-sdk-<version>.jar 
+  (only if any input event log is from S3)
+  
+## Download the tools jar or compile it
+You do not need to compile the jar yourself because you can download it from maven repository directly.
+
+Here are 2 options:
+1. Download the jar file from [maven repository](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark-tools_2.12/21.06.0/)
+
+2. Compile the jar from github repo
+```bash
+git clone https://github.com/NVIDIA/spark-rapids.git
+cd spark-rapids
+mvn -pl .,tools clean verify -DskipTests
+```
+The jar is generated in below directory :
+
+`./tools/target/rapids-4-spark-tools_2.12-<version>.jar`
+
+If any input is a S3 file path or directory path, 2 extra steps are needed to access S3 in Spark:
+1. Download the matched jars based on the Hadoop version:
+   - `hadoop-aws-<version>.jar`
+   - `aws-java-sdk-<version>.jar`
```

Comment:
I thought we needed an addition jar?

Reply:
Those two are good enough as per my tests for Hadoop 2.7.4. For Hadoop 3.x, they may need an extra jars but I have not tested. That is why we mention the doc link for Hadoop+AWS so that users can test themselves.